plenum/test/replica/stashing/test_stash_future_view.py

```diff
@@ -58,20 +54,12 @@ def test_process_three_phase_msg_and_stashed_future_view(txnPoolNodeSet, looper,
             # (len(txnPoolNodeSet) - 2) - prepare msgs
             # (len(txnPoolNodeSet) - 1) - commit msgs
             stashed_master_messages = 2 * (1 + (len(txnPoolNodeSet) - 2) + (len(txnPoolNodeSet) - 1))
-            stashed_backup_messages = 2 * (1 + (len(txnPoolNodeSet) - 2) + (len(txnPoolNodeSet) - 1))
             assert slow_node.master_replica.stasher.stash_size(STASH_VIEW_3PC) == old_stashed[0] + stashed_master_messages
-            assert all(r.stasher.stash_size(STASH_VIEW_3PC) == old_stashed[inst_id] + stashed_backup_messages
-                       for inst_id, r in slow_node.replicas.items() if inst_id != 0)
-            assert all(r.last_ordered_3pc == last_ordered[inst_id]
-                       for inst_id, r in slow_node.replicas.items())
 
         def chk():
             for inst_id, r in slow_node.replicas.items():
-                if inst_id == 0:
-                    assert r.last_ordered_3pc == (view_no + 1, 2)
-                else:
-                    assert r.last_ordered_3pc == (view_no + 1, 2)
-                assert r.stasher.stash_size(STASH_VIEW_3PC) == old_stashed[inst_id]
+                assert r.last_ordered_3pc[1] == (2 if r.isMaster else 1)
```

Comment:
Why is it 2 for master and 1 for backup?

Reply:
I suspect this is because primaries update batch after view change is sent only by master instance

Reply:
Should we send it by backups as well for symmetry in performance and code?

Reply:
From performance symmetry PoV - first reaction would be yes, but the problem is that currently on backups view change is finished immediately, so sending primaries batch at this point would make situation even more asymmetric. Probably we should synchronize view change finish on backups with master...

Reply:
Yes. I think it would be much more clear if backup follows the same sequence of steps/messages with the following deviations:
1) Checkpoint Service: process_new_view_accepted: Do not stabilize checkpoint from NewView, but reset ppSeqNo to 0 instead
2) Ordering service: process_new_view_checkpoints_applied: do not process any batches for re-order

BTW this is just thoughts, not must have actions for this PR

Reply:
> Checkpoint Service: process_new_view_accepted

But in this case checkpoint service will be modifying "foreign" fields (ppSeqNo), isn't it?

Also in this case we'll need to propagate `NewViewAccepted` from primary to backup replicas - not a problem in my opinion, just a notice

Reply:
> But in this case checkpoint service will be modifying "foreign" fields (ppSeqNo), isn't it
Well, we can do the rest in Ordering Service on `process_new_view_checkpoints_applied`